common/classes/com/arjuna/common/util/ConfigurationInfo.java

```diff
@@ -21,19 +21,48 @@
 package com.arjuna.common.util;
 
 import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
 
 /**
  * Utility class providing access to build time and runtime configuration reporting functions.
  *
  * Replaces the old per-module Info (and in some cases Configuration and report) classes.
  *
  * The actual build information is injected during the build via the
- * org.jboss.maven.plugins:maven-injection-plugin plugin
+ * Maven Resources Plugin
  *
  * @author Jonathan Halliday (jonathan.halliday@redhat.com) 2009-10
  */
 public class ConfigurationInfo
 {
+    private static final String SOURCE_ID;
+    private static final String PROPERTIES_FILE_NAME;
+    private static final String BUILD_ID;
+
+    static {
+        Properties configInfoProps = new Properties();
+        String sourceId = "(unknown)";
+        String propertiesFileName = "(unknown)";
+        String buildId = "(unknown)";
+        try (final InputStream stream = Thread.currentThread().getContextClassLoader().getResourceAsStream("ConfigurationInfo.properties")) {
```

Comment:
Why `Thread.currentThread().getContextClassLoader()`? Personally, I am always overcareful when it comes to Threads (in this case, `ConfigurationInfo` is used in ArjunaJTS -> JTS ... so I am even more worried (maybe too much lol)). Could you please help me understanding why you went for `Thread.currentThread().getContextClassLoader()`? Thanks

Reply:
I am not sure I follow you. If `A` and `B` are modules and `B` uses `A`, why should they run on different threads?

Reply:
Suppose module A is already built and now its module B's turn to get build and run. When B starts building at that time all the A's class and resource files package are ready to be used by module B. Now during the build `maven resource plugin filter` will be applied to B's resource file and then all the variables inside B's resource file will be replaces by their respective values and depending on the type of resource filter that is applied, will decide the location to resource file in the `/target` folder (which can be any of `/target/classes` or `/target/test-classes`)

Now its time to run the tests in B. First test in B tries to run. At this time there will be single `thread` that will be running. So the first test will try to call a class of A which has `Thread.currentThread().getContextClassLoader().getResourceAsStream("R")` in it. Here same test thread will try to get the resource file `R`. So instead of looking resource file into A's resource package, Thread will try to find the resource path of the module from where it started executing first, this means it will get the path of B's resource package.
You can verify this by checking the path value returned by `Thread.currentThread().getContextClassLoader().getResource("")`